modules/core/src/main/java/org/apache/synapse/core/axis2/TimeoutHandler.java

```diff
@@ -233,6 +233,8 @@ private void processCallbacks() {
                     if (RuntimeStatisticCollector.isStatisticsEnabled()) {
                         CallbackStatisticCollector.callbackCompletionEvent(callback.getSynapseOutMsgCtx(), (String) key);
                     }
+                    // Call the TransportSender's onAppError method to release any resources
+                    callback.getAxis2OutMsgCtx().getTransportOut().getSender().onAppError(callback.getAxis2OutMsgCtx());
                 }
```

Comment:
Can there be a NPE? if sender is not available for some reason? shall we handle those cases?

Reply:
The sending flow will break if the sender is not available and won't reach this. As a safety measure, I'll add a null check

Reply:
Fixed with 0328c2c